packages/eas-cli/src/commands/build/download.ts

```diff
@@ -0,0 +1,138 @@
+import { Platform } from '@expo/eas-build-job';
+import { Errors, Flags } from '@oclif/core';
+import chalk from 'chalk';
+import { pathExists } from 'fs-extra';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { ExpoGraphqlClient } from '../../commandUtils/context/contextUtils/createGraphqlClient';
+import { EasNonInteractiveAndJsonFlags } from '../../commandUtils/flags';
+import { AppPlatform, BuildFragment, BuildStatus, DistributionType } from '../../graphql/generated';
+import { BuildQuery } from '../../graphql/queries/BuildQuery';
+import Log from '../../log';
+import { promptAsync } from '../../prompts';
+import { getEasBuildRunCachedAppPath } from '../../run/run';
+import { downloadAndMaybeExtractAppAsync } from '../../utils/download';
+import { enableJsonOutput, printJsonOnlyOutput } from '../../utils/json';
+
+export default class Download extends EasCommand {
+  static override description = 'download builds for a given fingerprint hash';
+
+  static override flags = {
+    fingerprint: Flags.string({
+      description: 'Fingerprint hash of the build to download',
+      required: true,
+    }),
+    platform: Flags.enum<Platform.IOS | Platform.ANDROID>({
+      char: 'p',
+      options: [Platform.IOS, Platform.ANDROID],
+    }),
+    ...EasNonInteractiveAndJsonFlags,
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.LoggedIn,
+    ...this.ContextOptions.ProjectId,
+  };
+
+  async runAsync(): Promise<void> {
+    const {
+      flags: { json: jsonFlag, 'non-interactive': nonInteractive, platform, fingerprint },
+    } = await this.parse(Download);
+
+    const {
+      loggedIn: { graphqlClient },
+      projectId,
+    } = await this.getContextAsync(Download, {
+      nonInteractive,
+    });
+
+    if (jsonFlag) {
+      enableJsonOutput();
+    }
+
+    const selectedPlatform = await resolvePlatformAsync(nonInteractive, platform);
+    const build = await this.getBuildAsync(graphqlClient, projectId, selectedPlatform, fingerprint);
+    const buildPath = await this.getPathToBuildAppAsync(build, selectedPlatform);
+    if (jsonFlag) {
+      const jsonResults: { path?: string } = {};
+      if (buildPath) {
+        jsonResults.path = buildPath;
+      }
```

Comment:
I think as it stands `buildPath` will always return a non-empty string (or it will throw)?

Reply:
That's correct, fixed